roles/rabbitmq/templates/etc/serverspec/rabbitmq_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+require 'socket'
+
+files= ['shadow', 'passwd']
+files.each do |file|
+  describe file("/etc/#{file}") do
+    it { should contain 'rabbitmq'}
+  end
+end  
+
+describe command('cat /etc/nova/nova.conf|grep rabbit_password') do
+  its(:stdout) { should_not contain /(rabbit_password=\w{0,7})$/ }
+end
+
+describe command('ps -ef | grep /usr/lib/rabbitmq/bin/rabbitmq-server') do
+  its(:stdout) { should contain '/usr/lib/rabbitmq/bin/rabbitmq-server'}
+end 
+
+files = {'rabbitmq-env.conf'=> 600, 'enabled_plugins'=> 644}
+files.each do |file , mode|
+  describe file("/etc/rabbitmq/#{file}") do
+    it { should be_file }
+    it { should be_mode mode }
+  end
+end 
+
+
+hostname = Socket.gethostname
+files = ["rabbit@#{hostname}.log.1", "rabbit@#{hostname}-sasl.log", 'shutdown_err', 'shutdown_log', 'startup_err', 'startup_log']
+files.each do |file|
+  describe file("/var/log/rabbitmq/#{file}") do
+    it { should be_file }
+    it { should be_mode 644 }
+  end
+end 
+
+describe file('/etc/logrotate.d/rabbitmq-server') do
+  file_contents = [ '/var/log/rabbitmq/*.log {',
+         '  weekly',
+         '  missingok',
+         '  rotate 20',
+         '  compress',
+         '  delaycompress',
+         '  notifyempty',
+         '  sharedscripts',
+         '  postrotate',
+         '  /etc/init.d/rabbitmq-server rotate-logs > /dev/null',
+         '}']
+  file_contents.each do |file_line|
+    it { should contain file_line}
+  end
+end
+
+describe command('rabbitmqctl environment | grep default_user') do
+  output = "      {default_user,<<\"guest\">>},\n      {default_user_tags,[administrator]},\n"
+  its(:stdout) { should eq output}
+end   
+
+describe command('rabbitmqctl environment | grep log_levels') do
+  output = "      {log_levels,[{connection,info}]},\n"
+  its(:stdout) { should eq output}
+end 
+
+describe command('rabbitmqctl environment | grep log') do
+  output = ["     [{error_logger,tty},\n","      {error_logger,{file,\"/var/log/rabbitmq/rabbit@controller1.log\"}},\n",
+            "      {log_levels,[{connection,info}]},\n","      {sasl_error_logger,\n","          {file,\"/var/log/rabbitmq/rabbit@controller1-sasl.log\"}},\n",
+            "      {ssl_cert_login_from,distinguished_name},\n","          [{backlog,128},\n","     [{http_log_dir,none},\n {sasl,[{errlog_type,error},{sasl_error_logger,false}]},\n"].join()
```

Comment:
That certainly looks easier to follow, but screen scraping at its best. This will be very fragile.